// social-likes-next/lib/services/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _facebook = require('social-likes-next/lib/services/facebook');

Object.defineProperty(exports, 'facebook', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_facebook).default;
  }
});

var _odnoklassniki = require('social-likes-next/lib/services/odnoklassniki');

Object.defineProperty(exports, 'odnoklassniki', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_odnoklassniki).default;
  }
});

var _pinterest = require('social-likes-next/lib/services/pinterest');

Object.defineProperty(exports, 'pinterest', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_pinterest).default;
  }
});

var _plusone = require('social-likes-next/lib/services/plusone');

Object.defineProperty(exports, 'plusone', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_plusone).default;
  }
});

var _twitter = require('social-likes-next/lib/services/twitter');

Object.defineProperty(exports, 'twitter', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_twitter).default;
  }
});

var _vkontakte = require('social-likes-next/lib/services/vkontakte');

Object.defineProperty(exports, 'vkontakte', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_vkontakte).default;
  }
});

var _telegram = require('social-likes-next/lib/services/telegram');

Object.defineProperty(exports, 'telegram', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_telegram).default;
  }
});

var _linkedin = require('social-likes-next/lib/services/linkedin');

Object.defineProperty(exports, 'linkedin', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_linkedin).default;
  }
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
